WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationRepository.kt

```diff
@@ -105,6 +109,33 @@ class VariationRepository @Inject constructor(
         return !result.isError
     }
 
+    suspend fun bulkCreateVariations(
+        remoteProductId: Long,
+        variationCandidates: List<VariationCandidate>
+    ): RequestResult = withContext(dispatchers.io) {
+        productStore.batchGenerateVariations(
+            WCProductStore.BatchGenerateVariationsPayload(
+                selectedSite.get(),
+                remoteProductId,
+                variations = variationCandidates.map { candidate ->
+                    candidate.map { variantOption ->
+                        WCProductVariationModel.ProductVariantOption(
+                            id = variantOption.id,
+                            name = variantOption.name,
+                            option = variantOption.option
+                        )
```

Comment:
Here we can use the `asSourceModel()` from `VariantOption`
```suggestion
                        variantOption.asSourceModel()
```